src/modules/rtp_media_server/rms_media.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (C) 2017-2018 Julien Chavanton jchavanton@gmail.com
+ *
+ * This file is part of Kamailio, a free SIP server.
+ *
+ * Kamailio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version
+ *
+ * Kamailio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "../../core/mem/shm.h"
+#include "../../core/sr_module.h"
+#include "rtp_media_server.h"
+
+inline static void *ptr_shm_malloc(size_t size)
+{
+	return shm_malloc(size);
+}
+inline static void *ptr_shm_realloc(void *ptr, size_t size)
+{
+	return shm_realloc(ptr, size);
+}
+inline static void ptr_shm_free(void *ptr)
+{
+	shm_free(ptr);
+}
+
+typedef struct shared_global_vars
+{
+	MSFactory *ms_factory;
+	gen_lock_t lock;
+} shared_global_vars_t;
+
+static shared_global_vars_t *vars;
+
+MSFilterDesc *rms_ms_filter_descs[] = {&ms_alaw_dec_desc, &ms_alaw_enc_desc,
+		&ms_ulaw_dec_desc, &ms_ulaw_enc_desc, &ms_rtp_send_desc,
+		&ms_rtp_recv_desc, &ms_dtmf_gen_desc, &ms_volume_desc,
+		&ms_equalizer_desc, &ms_channel_adapter_desc, &ms_audio_mixer_desc,
+		&ms_tone_detector_desc, &ms_speex_dec_desc, &ms_speex_enc_desc,
+		&ms_speex_ec_desc, &ms_file_player_desc, &ms_file_rec_desc,
+		&ms_resample_desc,
+		//       &ms_opus_dec_desc,
+		//       &ms_opus_enc_desc,
+		NULL};
+
+static MSFactory *rms_create_factory()
+{
+	MSFactory *f = ms_factory_new();
+	int i;
+	for(i = 0; rms_ms_filter_descs[i] != NULL; i++) {
+		ms_factory_register_filter(f, rms_ms_filter_descs[i]);
+	}
+	ms_factory_init_plugins(f);
+	ms_factory_enable_statistics(f, TRUE);
+	ms_factory_reset_statistics(f);
+	return f;
+}
+
+int rms_media_init()
+{
+	OrtpMemoryFunctions ortp_memory_functions;
+	ortp_memory_functions.malloc_fun = ptr_shm_malloc;
+	ortp_memory_functions.realloc_fun = ptr_shm_realloc;
+	ortp_memory_functions.free_fun = ptr_shm_free;
+	ortp_set_memory_functions(&ortp_memory_functions);
+	ortp_init();
+	vars = ortp_malloc(sizeof(shared_global_vars_t));
```

Comment:
Why you use the ortp_malloc() for allocation of Kamailio module structures?
The malloc can also fail.

Reply:
no good reason, ortp_malloc is using shm_malloc, I replaced the 2 occurrences with shm_malloc for clarity from Kamailio module context